packages/edit-widgets/src/components/save-button/index.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * WordPress dependencies
+ */
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useDispatch } from '@wordpress/data';
+import { useState, useCallback } from '@wordpress/element';
+
+function SaveButton() {
+	const [ isSaving, setIsSaving ] = useState( false );
+	const { saveWidgetAreas } = useDispatch( 'core/edit-widgets' );
+	const onClick = useCallback( async () => {
+		setIsSaving( true );
```

Comment:
This made me wonder if we'd need a `isPerforming` selector for actions similar to the `isResolving` one we have for resolvers.

Reply:
I like this idea but wonder why not also name it `isResolving` since, to me, having an entity saved is a 'resolution'.

Reply:
> This made me wonder if we'd need a `isPerforming` selector for actions similar to the `isResolving` one we have for resolvers. cc @aduth

Built-in? Or as a pattern?

Generally I'd like if these could actually be resolvers, so we're not setting any state, the state is reflected by the selection of the existing `isResolving` on the widgets save resolution.

Reply:
I was thinking built-in similarily to resolvers but for actions. Technically speaking resolvers and actions are the exact same thing right now (in terms of code), the only different is that one get triggered when you call a selector, the other when you call an action :)